Xamarin.Build.AsyncTask/WaitForBackgroundTasks.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.IO;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using System.Threading;
+using System.Collections.Generic;
+using TPL = System.Threading.Tasks;
+
+namespace Xamarin.Build
+{
+    /// <summary>
+    /// This Task will wait for all the background tasks to complete.
+    /// You can control which task categories to wait on by using the
+    /// `Categories` property.
+    /// </summary>
+    public class WaitForBackgroundTasks : AsyncTask
+    {
+        /// <summary>
+        /// A list of background task categories to wait on.
+        /// Once all the tasks in each of these categories have Completed
+        /// the task will exit.
+        /// By default it will wait on the `default` category.
+        /// </summary>
+        public string [] Categories { get; set; }
+
+        /// <summary>
+        /// The error code to use should any of the background tasks fail.
+        /// this will default to XAT0000
+        /// </summary>
+        public string ErrorCode { get; set; }
```

Comment:
```suggestion
        public string ErrorCode { get; set; } = "XAT0000";
```
Then you could remove the `ctor`.